tests/testthat/test-cube-subset.R

```diff
@@ -0,0 +1,297 @@
+            )
+        )
+    )
+    expect_identical(subsetArrayDimension(cat_x_mr_x_mr@dims[[1]], 1:2), expected)
+})
+
+test_that("subsetArrayDimension MR dimension", {
+    expected <- list(
+        name = c("rest_opinion","play_opinion"),
+        any.or.none = c(FALSE, FALSE),
+        missing = c(FALSE, FALSE),
+        references = list(
+            description = "",
+            format = list(
+                summary = list(digits = 0L)
+            ),
+            subreferences = list(
+                list(alias = "rest_opinion#", name = "rest_opinion"),
+                list(alias = "play_opinion#", name = "play_opinion")
+            ),
+            notes = "",
+            name = "opinion MR",
+            discarded = FALSE,
+            alias = "opinion_mr",
+            view = list(
+                show_counts = FALSE,
+                include_missing = FALSE,
+                column_width = NULL
+            ),
+            type = "subvariable_items"
+        )
+    )
+    expect_identical(subsetArrayDimension(cat_x_mr_x_mr@dims[[2]], 2:3), expected)
+})
+
+test_that("subsetArrayDimension categorical array dimension", {
+    expected <- list(
+        name = c("cat_feeling"),
+        any.or.none = c(FALSE),
+        missing = c(FALSE),
+        references = list(
+            subreferences = list(
+                list(alias = "cat_feeling", name = "cat_feeling")
+            ),
+            name = "feeling CA",
+            alias = "feeling_ca",
+            type = "subvariable_items"
+        )
+    )
+    expect_identical(subsetArrayDimension(catarray_x_mr@dims[[1]], 1), expected)
+
+    expected <- list(
+        name = c("Somewhat Happy", "Neutral"),
+        any.or.none = c(FALSE, FALSE),
+        missing = c(FALSE, FALSE),
+        references = list(
+            subreferences = list(
+                list(alias = "cat_feeling", name = "cat_feeling"),
+                list(alias = "dog_feeling", name = "dog_feeling")
+            ),
+            name = "feeling CA",
+            alias = "feeling_ca",
+            type = "categorical",
+            categories = list(
+                list(numeric_value = 2L,
+                     id = 2L,
+                     name = "Somewhat Happy",
+                     missing = FALSE),
+                list(numeric_value = 3L,
+                     id = 3L,
+                     name = "Neutral",
+                     missing = FALSE)
+            )
+        )
+    )
+    expect_identical(subsetArrayDimension(catarray_x_mr@dims[[2]], c(2:3)), expected)
+})
+
+test_that("translateCubeIndex", {
+    expect_identical(translateCubeIndex(cat_x_mr_x_mr, alist(1, ,), drop = FALSE),
+        alist(1, , TRUE, , TRUE))
+    expect_identical(translateCubeIndex(cat_x_mr_x_mr, alist(1:2, 1:2, 1:2), drop = FALSE),
+        alist(1:2, 1:2, TRUE, 1:2, TRUE))
+    expect_identical(translateCubeIndex(cat_x_mr_x_mr, alist(1:2, 1:2, 1:2), drop = FALSE),
+        alist(1:2, 1:2, TRUE, 1:2, TRUE))
```

Comment:
I might be missing something subtle here, but are line 112-113 the same as 114-115?

Reply:
You miss nothing :)